src/dotnet/commands/dotnet-msbuild/MSBuildForwardingApp.cs

```diff
@@ -77,6 +79,11 @@ internal static CommandOption AddVerbosityOption(CommandLineApplication app)
             return app.Option("-v|--verbosity", LocalizableStrings.VerbosityOptionDescription, CommandOptionType.SingleValue);
         }
 
+        private static string Escape(string arg) =>
+             (arg.StartsWith("/p:RestoreSources=", StringComparison.OrdinalIgnoreCase)) ?
+                arg.Replace(";", "%3B") : // <-- this is a workaround for https://github.com/Microsoft/msbuild/issues/1622
```

Comment:
I'm afraid that without https://github.com/Microsoft/msbuild/pull/1641 you'd also have to escape the forward slashes. In the repro, the build still succeeds. If you do a diagnostic log, you'd see:

```
_GenerateRestoreGraph: (TargetId:4)
                     Output Item(s): 
                         ...
                                Sources=/tmp/some-source;https:/api.nuget.org/v3/index.json
                         ...
```

I don't know why it succeeds maybe nuget swallows the error.